torch_np/tests/numpy_tests/core/test_multiarray.py

```diff
@@ -2667,14 +2667,10 @@ def test_put(self):
             a.put([1, 3, 5], [True]*3)
             assert_equal(a, tgt.reshape(2, 3))
 
-        # check must be writeable
-        a = np.zeros(6)
-        a.flags.writeable = False
-        assert_raises(ValueError, a.put, [1, 3, 5], [1, 3, 5])
-
         # when calling np.put, make sure a
         # TypeError is raised if the object
         # isn't an ndarray
+        pytest.xfail("XXX: Argument normalisation prevents catching this")
```

Comment:
Does it mean that it needs to be annotated as NDArray, not ArrayLike?

Reply:
Ah I see yeah, thanks.